src/formatDate.js

```diff
@@ -8,7 +8,85 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const normilizedFromFormat = formatDestructing(fromFormat);
+  const normilizedToFormat = formatDestructing(toFormat);
+
+  // let dateArr = date.split(normilizedFromFormat.delim);
+
+  let day;
+  let month;
+  let year;
+
+  if (normilizedFromFormat.format === 'DMY') {
+    [day, month, year] = date.split(normilizedFromFormat.delim);
+  } else if (normilizedFromFormat.format === 'DYM') {
+    [day, year, month] = date.split(normilizedFromFormat.delim);
+  } else if (normilizedFromFormat.format === 'YDM') {
+    [year, day, month] = date.split(normilizedFromFormat.delim);
+  } else if (normilizedFromFormat.format === 'YMD') {
+    [year, month, day] = date.split(normilizedFromFormat.delim);
+  } else if (normilizedFromFormat.format === 'MDY') {
+    [month, day, year] = date.split(normilizedFromFormat.delim);
+  } else if (normilizedFromFormat.format === 'MYD') {
+    [month, year, day] = date.split(normilizedFromFormat.delim);
+  }
```

Comment:
The series of 'if' statements for splitting the date can be simplified. Consider using an object map to look up the date parts based on the format string, which can help reduce the repetitive code and improve maintainability.